src/runtime/internal/dom.ts

```diff
@@ -640,6 +640,24 @@ export function query_selector_all(selector: string, parent: HTMLElement = docum
 	return Array.from(parent.querySelectorAll(selector)) as ChildNodeArray;
 }
 
+export function head_selector(nodeId: string, head: HTMLElement) {
+	const result = [];
+	let started = 0;
+
+	for (const node of head.childNodes) {
+		if (node.nodeType === 8 /* comment node */ && node.textContent.trim() === `HEAD_END data-svelte="${nodeId}"`) {
+			started -= 1;
+			result.push(node);
+		} else if (node.nodeType === 8 /* comment node */ && node.textContent.trim() === `HEAD_START data-svelte="${nodeId}"`) {
+			started += 1;
+			result.push(node);
+		} else if (started > 0) {
+			result.push(node);
+		}
```

Comment:
can refactor to something like this, would make it cleaner, without having to check for `node.nodeType` and doing `node.textContent.trim()` twice

```suggestion
		if (node.nodeType === 8 /* comment node */) {
			const comment = node.textContent.trim();
			if (comment === `HEAD_END data-svelte="${nodeId}"`) {
				started -= 1;
				result.push(node);
				continue;
			} else if (comment === `HEAD_START data-svelte="${nodeId}"`) {
				started += 1;
				result.push(node);
				continue;
			}
		}
		result.push(node);
```